src/_pytest/_code/code.py

```diff
@@ -479,7 +479,7 @@ def from_current(
         assert tup[1] is not None, "no current exception"
         assert tup[2] is not None, "no current exception"
         exc_info = (tup[0], tup[1], tup[2])
-        return cls.from_exc_info(exc_info, exprinfo)
+        return ExceptionInfo.from_exc_info(exc_info, exprinfo)
```

Comment:
This is needed because it seems now `cls` is bound to the `_E` generic which is not compatible with `BaseException`. Using the type directly "unbounds" it.